apps/browser/src/autofill/background/context-menus.background.ts

```diff
@@ -20,17 +20,16 @@ export default class ContextMenusBackground {
 
     BrowserApi.messageListener(
       "contextmenus.background",
-      async (
+      (
         msg: { command: string; data: LockedVaultPendingNotificationsItem },
-        sender: chrome.runtime.MessageSender,
-        sendResponse: any
+        sender: chrome.runtime.MessageSender
       ) => {
         if (msg.command === "unlockCompleted" && msg.data.target === "contextmenus.background") {
-          await this.contextMenuClickedHandler.cipherAction(
-            msg.data.commandToRetry.msg.data,
-            msg.data.commandToRetry.sender.tab
-          );
-          await BrowserApi.tabSendMessageData(sender.tab, "closeNotificationBar");
+          this.contextMenuClickedHandler
+            .cipherAction(msg.data.commandToRetry.msg.data, msg.data.commandToRetry.sender.tab)
+            .then(() => {
+              BrowserApi.tabSendMessageData(sender.tab, "closeNotificationBar");
+            });
```

Comment:
question(non-blocking): why are changing from async/await to promise-then syntax? I usually think that the former is easier to read and more consistent with the majority of the code-base

Reply:
@coroiu 

I agree with your sentiment here. What would make sense is to pull this behavior out to a separate method so we could still consider the async/await approach.

That suggestion aside, I advised Jason on these changes and can give a reason for why, in this instance, we're avoiding using the `await` keyword.

When using the `chrome.runtime.onMessage.addListener` API, any callback that is passed to the listener needs to be structured as a syncrhonous callback. This is due to how the `sendMessage` parameter is structured to respond when a message from `chrome.runtime.sendMessage` is received. 

The issue that we were seeing in this ticket was occurring because we were setting up the callbacks for `onMessage.addListener` as `async` methods, which by default return a `Promise<void>`. Because of that, we were experiencing race conditions where some `chrome.runtime.sendMessage` calls that were expecting a response from the `onMessage` listener were receiving `undefined` values. 

This MDN article explains a bit more on the issue - https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/runtime/onMessage#parameters

I think if @Jingo88 were to pull this behavior out to a private method, they could then place the method in the `onMessage` callback function without having to use a `.then` chain.

Reply:
```suggestion
          handleUnlockCompletedCommand();
            
            [...]
            
            async handleUnlockCompletedCommand() {
                await this.contextMenuClickedHandler.cipherAction(msg.data.commandToRetry.msg.data, msg.data.commandToRetry.sender.tab);
               BrowserApi.tabSendMessageData(sender.tab, "closeNotificationBar");
            }
```

Something akin to this.

Reply:
Ohhhhhhhh of course! I never occured to me that making the functions `async` would cause them to return a `Promise` behind the scenes, which would be a truthy value. Someone if 100% going to fall for this again and having no idea what the issue is! Could we maybe force the callbacks to return `boolean | void` and add a comment explaining why `Promises` are forbidden?

Reply:
I think that would be would be a solid idea.

Reply:
Leaving open to make it easier to find in the future